packages/react/src/components/Toolbar/Toolbar.tsx

```diff
@@ -118,189 +97,348 @@ class Toolbar extends UIComponent<WithAsProp<ToolbarProps>, ToolbarState> {
-                          wrapper,
-                        ),
-                      },
-                      overrideProps: () => ({
-                        children: (
-                          <Ref innerRef={this.hiddenToolbarRef}>
-                            <ElementType
-                              className={classes.root}
-                              {...accessibility.attributes.root}
-                              {...unhandledProps}
-                            >
-                              {this.renderItems(this.state.currentItems, variables)}
-                            </ElementType>
-                          </Ref>
-                        ),
-                      }),
-                    })}
-                  {this.state.stableItems && (
-                    <ElementType
-                      className={classes.root}
-                      {...accessibility.attributes.root}
-                      {...unhandledProps}
-                    >
-                      {this.renderItems(this.state.stableItems, variables)}
-                    </ElementType>
-                  )}
-                  <ReactResizeDetector xskipOnMount handleWidth onResize={this.onResize} />
-                </>
-              ),
-            }),
-          })}
-        </Ref>
+        <ElementType
+          className={classes.root}
+          {...accessibility.attributes.root}
+          {...unhandledProps}
+        >
+          {childrenExist(children) ? children : this.renderItems(items, variables)}
+        </ElementType>
       )
     }
 
     return (
-      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
-        {childrenExist(children) ? children : this.renderItems(items, variables)}
-      </ElementType>
+      <>
+        <ElementType
+          className={classes.root}
+          {...accessibility.attributes.root}
+          {...unhandledProps}
+        >
+          <Ref innerRef={this.overflowContainerRef}>
+            {Box.create(overflowContainer, {
+              defaultProps: {
+                styles: styles.overflowContainer,
+              },
+              overrideProps: {
+                children: (
+                  <>
+                    {childrenExist(children) ? children : this.renderItems(items, variables)}
+                    <Ref innerRef={this.overflowItemRef}>
+                      <ToolbarOverflowMenu getOverflowItems={this.getOverflowItems} />
+                    </Ref>
+                  </>
+                ),
+              },
+            })}
+          </Ref>
+          <Ref innerRef={this.offsetMeasureRef}>
+            {Box.create(offsetMeasure, {
+              defaultProps: {
+                styles: styles.offsetMeasure,
+              },
+            })}
+          </Ref>
+        </ElementType>
+        <EventListener
+          listener={_.debounce(this.hideOverflowItems, 16)}
+          targetRef={windowRef}
+          type="resize"
```

Comment:
Is it enough to listen for `resize`? What about `scroll`?

Reply:
Should be